python/ql/src/experimental/Security/CWE-295/UrlopenWithoutValidation.ql

```diff
@@ -0,0 +1,44 @@
+/**
+ * @name Attempt to open url without certificate validation
+ * @description Opening a url without certificate validation can allow man-in-the-middle attacks.
+ * @kind problem
+ * @problem.severity error
+ * @precision medium
+ * @id py/urlopen-without-cert-validation
+ * @tags security
+ *       external/cwe/cwe-295
+ */
+
+import python
+
+from
+  FunctionValue urlopen, CallNode call, ControlFlowNode http_arg, StringValue http_string,
+  Value empty
+where
+  (
+    urlopen = Value::named("urllib.request.urlopen")
+    or
+    urlopen = Value::named("urllib2.request.urlopen")
+  ) and
```

Comment:
I think you could refactor that to use [set-literals](https://help.semmle.com/QL/ql-handbook/expressions.html#set-literal-expressions) instead.
Like this:
```suggestion
  urlopen = Value::named(["urllib2.request.urlopen", "urllib.request.urlopen"]) and
```

Reply:
Thanks for the tip!